eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/steps/processors/ContextNucleotideAdditionProcessor.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.eva.accession.release.steps.processors;
+
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.accession.dbsnp.io.FastaSynonymSequenceReader;
+import uk.ac.ebi.eva.commons.core.models.IVariant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+
+import static java.lang.Math.max;
+
+public class ContextNucleotideAdditionProcessor implements ItemProcessor<IVariant, IVariant> {
+
+    private FastaSynonymSequenceReader fastaSequenceReader;
+
+    public ContextNucleotideAdditionProcessor(FastaSynonymSequenceReader fastaReader) {
+        this.fastaSequenceReader = fastaReader;
+    }
+
+    @Override
+    public IVariant process(final IVariant variant) throws Exception {

```

Comment:
Are we sure we want to make this object final and create a new one for every insertion? I am a bit concerned that if we follow this strategy in every single processor we will create tons of short-lived objects. We don't do it in any other unless it vastly simplifies the logic.